edbdeploy/project.py

```diff
@@ -287,7 +287,97 @@ def _build_terraform_vars_file(self, env):
+                },
+                'hammerdb': ra['hammerdb'],
+                'hammerdb_server': {
+                    'count': 1 if ra['hammerdb_server'] else 0,
+                    'instance_type': hammerdb['instance_type'],
+                    'volume': hammerdb['volume'],
+                },
+                'pem_server': {
+                    'count': 1 if ra['pem_server'] else 0,
+                    'instance_type': pem['instance_type'],
+                    'volume': pem['volume'],
+                },
+                'pg_version': env.postgres_version,
+                'pooler_local': ra['pooler_local'],
+                'pooler_type': ra['pooler_type'],
+                'postgres_server': {
+                    'count': ra['pg_count'],
+                    'instance_type': pg['instance_type'],
+                },
+                'pg_type': env.postgres_type,
+                'replication_type': ra['replication_type'],
+                'ssh_pub_key': self.ssh_pub_key,
+                'ssh_priv_key': self.ssh_priv_key,
+                'ssh_user': os['ssh_user'],
+            }
+
+            if 'barman_server' in env.cloud_spec:
+                barman = env.cloud_spec['barman_server']
+                self.terraform_vars.update({
+                    'barman_server': {
+                        'count': ra['barman_server_count'],
+                        'instance_type': barman['instance_type'],
+                        'volume': barman['volume'],
+                        'additional_volumes':
+                                barman['additional_volumes'],
+                    },
+                })
+
+            if 'bdr_server' in env.cloud_spec:
+                bdr = env.cloud_spec['bdr_server']
+                self.terraform_vars.update({
+                    'bdr_server': {
+                        'count': ra['bdr_server_count'],
+                        'instance_type': bdr['instance_type'],
+                        'volume': bdr['volume'],
+                        'additional_volumes': bdr['additional_volumes'],
+                    },
+                })
+
+            if 'bdr_witness_server' in env.cloud_spec:
+                bdr_witness = env.cloud_spec['bdr_witness_server']
+                self.terraform_vars.update({
+                    'bdr_witness_server': {
+                        'count': ra['bdr_witness_count'],
+                        'instance_type': bdr_witness['instance_type'],
+                        'volume': bdr_witness['volume'],
+                    },
+                })
+
+            if 'pooler_server' in env.cloud_spec:
+                pooler = env.cloud_spec['pooler_server']
+                self.terraform_vars.update({
+                    'pooler_server': {
+                        'count': ra['pooler_count'],
+                        'instance_type': pooler['instance_type'],
+                        'volume': pooler['volume'],
+                    },
+                })
+
```

Comment:
I don't think `_build_terraform_vars_file()` is the right location to configure terraform variables: if we go this way, we need to update `projects/aws_pot.py`, `projects/gcloud_pot.py`, etc... all the classes that implement `_build_terraform_vars()` must be updated, because they are all impacted by the changes applied on `_build_terraform_vars_files()`.

Here is my suggestion, if we want to do this refactoring work incrementally, first with `aws`, `azure` and `gcloud`, and then cover all the other cases later in an other PR:

1. move this code into a new method, let say `_init_terraform_vars()`
2. in `projects/aws.py`, `projects/azure.py` and `projects/gcloud.py`:
```python
    def _build_terraform_vars(self):
        # Initialize terraform variables with common values
        self._init_terraform_vars()
        # Configure specific terraform variables
        self.terraform_vars.update({
           ....
        })
```
This way, we won't break anything in the classes that are not yet refactored.

Reply:
That make sense.  I've push another set of changes to do that.